joblib/test/test_parallel.py

```diff
@@ -166,6 +168,30 @@ def test_simple_parallel():
         yield check_simple_parallel, backend
 
 
+def check_main_thread_renamed_no_warning(backend):
+    with warnings.catch_warnings(record=True) as caught_warnings:
+        warnings.simplefilter("always")
+        results = Parallel(n_jobs=2, backend=backend)(
+            delayed(square)(x) for x in range(3))
+        assert_equal(results, [0, 1, 4])
+    assert_equal(caught_warnings, [])
```

Comment:
Rather than checking for no warning (took me a while to figure out why we were doing it). Could we not do something like:
```python
p = Parallel(n_jobs=2, backend=backend)
assert_equal(p._backend.configure(), 2)
```

There may be a more elegant way of doing it.